// gatsby-plugin-netlify/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PAGE_COUNT_WARN = exports.HEADER_COMMENT = exports.ROOT_WILDCARD = exports.LINK_REGEX = exports.CACHING_HEADERS = exports.IMMUTABLE_CACHING_HEADER = exports.SECURITY_HEADERS = exports.DEFAULT_OPTIONS = exports.NETLIFY_HEADERS_FILENAME = exports.BUILD_BROWSER_BUNDLE_STAGE = void 0;
// Gatsby values
exports.BUILD_BROWSER_BUNDLE_STAGE = `build-javascript`;
// Plugin values
exports.NETLIFY_HEADERS_FILENAME = `_headers`;
// Default options including transform to manipulate headers for
// sorting and rewrites for client only paths
exports.DEFAULT_OPTIONS = {
    headers: {},
    mergeSecurityHeaders: true,
    mergeLinkHeaders: true,
    mergeCachingHeaders: true,
    transformHeaders: (value) => value,
    generateMatchPathRewrites: true,
};
exports.SECURITY_HEADERS = {
    '/*': [
        `X-Frame-Options: DENY`,
        `X-XSS-Protection: 1; mode=block`,
        `X-Content-Type-Options: nosniff`,
        `Referrer-Policy: same-origin`,
    ],
};
exports.IMMUTABLE_CACHING_HEADER = `Cache-Control: public, max-age=31536000, immutable`;
exports.CACHING_HEADERS = {
    '/static/*': [exports.IMMUTABLE_CACHING_HEADER],
    '/sw.js': [`Cache-Control: no-cache`],
};
exports.LINK_REGEX = /^(Link: <\/)(.+)(>;.+)/;
exports.ROOT_WILDCARD = `/*`;
exports.HEADER_COMMENT = `## Created with gatsby-plugin-netlify`;
exports.PAGE_COUNT_WARN = 1000;
